drivers/hmis/app/graphql/mutations/delete_service_type_type.rb

```diff
@@ -0,0 +1,27 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Mutations
+  class DeleteServiceTypeType < BaseMutation
+    argument :id, ID, required: true
+
+    field :service_type, Types::HmisSchema::ServiceType, null: true
+
+    def resolve(id:)
+      raise 'access denied' unless current_user.can_configure_data_collection?
+
+      service_type = Hmis::Hud::CustomServiceType.find_by(id: id)
+      raise HmisErrors::ApiError, 'Invalid service type ID' unless service_type
+
+      is_empty = service_type.custom_services.count == 0
+      raise HmisErrors::ApiError, 'Cannot delete a service type that has services' unless is_empty
```

Comment:
"exists?" does a faster query than `count`
```suggestion
      raise 'Cannot delete a service type that has services' if service_type.custom_service.exists?
```